ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/participantProfileServiceAgent.service.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { forkJoin, of } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+import { ConfigurationService, LoggingService, ParticipantProfileServiceAgent, SessionMementoService, UserProfile } from 'ddp-sdk';
+
+describe('ParticipantProfileServiceAgent', () => {
+    let service: ParticipantProfileServiceAgent;
+    let httpClient: HttpClient;
+    let httpTestingController: HttpTestingController;
+    let config: ConfigurationService;
+    const backendUrl = 'https://pepper-dev.datadonationplatform.org';
+
+    beforeEach(() => {
+        config = new ConfigurationService();
+        config.backendUrl = backendUrl;
+        config.studyGuid = 'ATCP';
+        const loggingServiceSpy: jasmine.SpyObj<LoggingService> = jasmine.createSpyObj('LoggingService', ['logException']);
+        const sessionSpy = new SessionMementoService({} as TranslateService, config);
+        spyOnProperty(sessionSpy, 'sessionObservable').and.returnValue(of({ participantGuid: 'USERGUID' }));
+
+        TestBed.configureTestingModule({
+            imports: [HttpClientTestingModule]
+        });
+
+        httpClient = TestBed.inject(HttpClient);
+        httpTestingController = TestBed.inject(HttpTestingController);
+
+        service = new ParticipantProfileServiceAgent(sessionSpy, config, httpClient, loggingServiceSpy);
+    });
+
+    afterEach(() => {
+        httpTestingController.verify();
+    });
+
+    it('Service should be initialized', () => {
+        expect(service).not.toBeNull();
+    });
+
+    it('should update participant profiles', (done) => {
+        const participantGuid1 = 'participantGuid1';
+        const participantGuid2 = 'participantGuid2';
+        const participants = [
+            {
+                guid: participantGuid1,
+                profile: {
+                    preferredLanguage: 'en'
+                } as UserProfile
+            },
+            {
+                guid: participantGuid2,
+                profile: {
+                    preferredLanguage: 'en',
+                    firstName: undefined,
+                    lastName: undefined
+                } as UserProfile
+            }
+        ];
+
+        forkJoin(
+            service.updateParticipantProfiles(participants)
+        ).subscribe((result: any[]) => {
+            expect(result.length).toEqual(2);
+            done();
+        });
+
+        const req1 = httpTestingController.expectOne(`${backendUrl}/pepper/v1/user/${participantGuid1}/profile`);
+        expect(req1.request.method).toBe('PATCH');
```

Comment:
I would rather have one assertion per test.

> “One assertion per test” is a wise rule to keep in mind, because it helps you have tests that fail for a specific reason, and drives you to focus on a specific behavior at a time.

[One assertion per test](https://pierodibello.medium.com/one-assertion-per-test-732cc2a7d3d#:~:text=%E2%80%9COne%20assertion%20per%20test%E2%80%9D%20is,specific%20behavior%20at%20a%20time.)
[Angular Unite Testing Best Practices](https://www.youtube.com/watch?v=7dcSpinbgqM&t=1323s)

Reply:
There is another [opinion](https://stackoverflow.blog/2022/11/03/multiple-assertions-per-test-are-fine/), though. :D . 
As this case is specific due to HttpClient, you can stick to whichever you like

Reply:
This [article](https://testing-angular.com/testing-services/#alternatives-for-finding-pending-requests) could be helpful for avoiding multiple assertions in this particular case by inserting some configuration into the `expectOne` method